### DISCUSSION TITLE
Visual Studio compatibility?

### DISCUSSION THREAD

**Discussion by Gibson85** [Ericsson/codechecker#3208] [Q&A] [answered]
Hi,

I would like to know if it is possible to use CodeChecker for Visual Studio solutions too?
I don't want to set up a make/cmake file for every project only for static test analysis.
From your example it seems that CodeChecker works together with make only?

^^^
CodeChecker check -b "cd ~/your-project && make clean && make" -o ~/results
^^^

Or maybe it could be possible to provide a folder to scan for *.hpp/*.cpp files to check?

**Comment by gyorb** [comment, marked as answer]
Hi,

CodeChecker depens on a [JSON Compilation Database](https://clang.llvm.org/docs/JSONCompilationDatabase.html) to run the analysis.
The `check` example executes the compilation, collects the compile commands and runs the analysis.

If your build system can create a compile command database CodeChecker should be able to use it.
cmake can export the compilation commands like this: `cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON`

I do not know if it is possible to get/export the compilation commands from Visual Studio somehow.

I found an extension by [sourcetrail](https://www.sourcetrail.com/blog/export_clang_compilation_database_from_visual_studio_solution/) but I never tried it.
It is open source and developed [here](https://github.com/CoatiSoftware/vs-sourcetrail) on github.

I'm not sure if there is a built-in feature in Visual Studio to get the compilation command database.
